munit-scalacheck/shared/src/main/scala/munit/ScalaCheckSuite.scala

```diff
@@ -19,23 +22,46 @@ trait ScalaCheckSuite extends FunSuite {
     test(options)(body)
   }
 
-  override def munitValueTransforms: List[ValueTransform] =
-    super.munitValueTransforms :+ scalaCheckPropTransform
+  // Allow property bodies of type Unit
+  // This is done to support using MUnit assertions in property bodies
+  // instead of returning a Boolean.
+  implicit def unitToProp: Unit => Prop = _ => Prop.passed
+
+  override def munitTestTransforms: List[TestTransform] =
+    super.munitTestTransforms :+ scalaCheckPropTransform
 
   protected def scalaCheckTestParameters = ScalaCheckTest.Parameters.default
 
   protected def scalaCheckPrettyParameters = Pretty.defaultParams
 
-  private def scalaCheckPropTransform: ValueTransform =
-    new ValueTransform("ScalaCheck Prop", {
-      case prop: Prop =>
-        val result = ScalaCheckTest.check(scalaCheckTestParameters, prop)
-        val prettyResult = Pretty.pretty(result, scalaCheckPrettyParameters)
-        if (result.passed) {
-          println(prettyResult)
-          Future.successful(())
-        } else {
-          Future.failed(new ScalaCheckFailException("\n" + prettyResult))
-        }
+  private val scalaCheckPropTransform: TestTransform =
+    new TestTransform("ScalaCheck Prop", t => {
+      t.withBodyMap[TestValue](
+        _.transformCompat {
+          case Success(prop: Prop) => propToTry(prop, t.location)
+          case r                   => r
+        }(munitExecutionContext)
+      )
     })
+
+  private def propToTry(prop: Prop, testLocation: Location): Try[Unit] = {
+    import ScalaCheckTest._
+    val result = check(scalaCheckTestParameters, prop)
+    def renderResult(r: Result) =
+      Pretty.pretty(r, scalaCheckPrettyParameters)
+
+    result.status match {
+      case Passed | Proved(_) =>
+        println(renderResult(result))
```

Comment:
@gabro just noticed this, is it intentional?

Reply:
Yes, it is. We send the output of successful tests to stdout

Reply:
Ah gotcha! 👍